### DISCUSSION TITLE
To Find an open IModelDb by its key.

### DISCUSSION THREAD

**Discussion by Adu2120** [iTwin/itwinjs-core#4333] [Q&A]
Where to get this key for findByKey() function of SnapshotDb to create an instance of SnapshotDb / IModelDb? Is there any another way to create the instance of IModelDb? 
[Click here for Documentation Link](https://www.itwinjs.org/reference/core-backend/imodels/snapshotdb/)

**Comment by pmconne** [comment]
`findByKey` does not create an IModelDb. It looks up one that was already created.
The key is [supplied or generated](https://www.itwinjs.org/reference/core-common/imodels/opendbkey/) when opening the iModel, e.g., by calling [SnapshotDb.openFile](https://www.itwinjs.org/reference/core-backend/imodels/snapshotdb/openfilestatic/).
